Automattic/wp-calypso
# Fix the transfer in domain status screen to use the new navigation component

#### Changes proposed in this Pull Request

* Turns out we didn't use the new domain navigation component in the transfer in status screen. While fixing that I noticed that we didn't have the correct routes set up for transfer ins so I've updated those too. We need to do that in order to clean up the unused navigation component.
* It's now safe to clean up the old navigation component too

#### Testing instructions

* Open an incoming domain transfer in Calypso and verify that you can see the cancel menu item if the transfer is pending. Otherwise no menu should be shown. Also verify that if you visit the transfer via the all domains view you'll see the "Manage your site" item.
